### ISSUE TITLE
Migrate from json to database

### ISSUE THREAD

**Issue by athityakumar** [metakgp/naarad-source#45]

**Comment by athityakumar** [comment]
It's better to populate the posts' data into database (MySQL/Postgre) - so let's decide how to continue -

(1) Jekyll
(2) Php
(3) Ruby on Rails
(4) Py + Django
(5) Py + Flask
(6) Python + sql-reading library + static-page creation (like current `frontend.py`)

Discuss!

cc @DefCon-007 @icyflame @hargup @vivekiitkgp @AvijitGhosh82

**Comment by icyflame** [comment]
@athityakumar I think just using Jekyll with proper Yaml front matter would be the simplest and most effective way to go. 

We have a script that will convert each post to a `md` file in the `_posts/` directory on the gh-pages branch. We can write templates, and all that. The deployment would be a breeze which would be a major positive.

The options from 3-6 are hard to deploy. PHP is not a good choice for maintainability concerns. (Some people in the future might not even know where to start! :-1: )

**Comment by athityakumar** [comment]
@icyflame : Yeah, it's easy to manage with Jekyll. But are you sure that the `_post/` directory won't be overloaded or something? Should we use Git LFS?

**Comment by icyflame** [comment]
@athityakumar `_posts/` won't be overloaded with a few thousand posts. Git LFS is for _large files_, not for _large number of files_. You need not worry.

**Comment by athityakumar** [comment]
@icyflame : But if we're including every post as a .md file in _posts , we're very well flooding the repo with large number of files. Right?

**Comment by icyflame** [comment]
@athityakumar It will be only on the `gh-pages` branch.

**Comment by athityakumar** [comment]
@icyflame : Great, so we're going ahead with Jekyll?

**Comment by icyflame** [comment]
@athityakumar That looks like the best idea to me. Though, you should think about what the difference between present setup and that would be. I don't see much of an improvement because we are still doing the same commit and then push and then render from GitHub business.

**Comment by athityakumar** [comment]
@icyflame : Yeah, but at least with Jekyll, we can easily implement pagination + read more / less with a unique page for each post. That way, we can show users a "preview" rather than  showing the full post, and pagination will also decrease the loading time of naarad.

**Comment by icyflame** [comment]
@athityakumar Why not just move from static-generation to a DB-based application in a single fell swoop? That would save a lot of time, and the time to implement both will be comparable. (We _could_ host this on the metakgp server, but you would have to figure out deployment. So, as an easier option you could go for Heroku)

cc @vivekiitkgp @amrav

**Comment by DefCon-007** [comment]
I think deploying on heroku will be great.

**Comment by athityakumar** [comment]
Great, so - not Jekyll. Then, "python library for sql & rendering of pages by frontend.py" - I choose you! :Smile:
